### PR TITLE
Added more types of encoder

### PR DESCRIPTION
**Pull request by Dishebh**
#### Changes done:
- [x] Added more types of encoder

Preview Link: https://deploy-preview-314--tender-ardinghelli-ebb79f.netlify.app/docs/MSI/encoder.html

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add a link from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

**Comment by Shivansh2407** [comment]
Please check the link

**Comment by Dishebh** [comment]
> Please check the link

Done, thanks for notifying.

File: docs/MSI/encoder.md [hunk 1]
```diff
@@ -50,4 +50,21 @@ That means if D3 = 1 then Y1 Y1 = 11 irrespective of the other inputs. Similarly
 
 <iframe width="100%" height="400px" src="https://circuitverse.org/simulator/embed/762" id="projectPreview" scrolling="no" webkitAllowFullScreen mozAllowFullScreen allowFullScreen> </iframe>
 
+## Decimal to BCD Encoder
+{: .no_toc }
+
+The decimal to binary encoder usually consists of 10 input lines and 4 output lines. Each input line corresponds to the each decimal digit and 4 outputs correspond to the BCD code. This encoder accepts the decoded decimal data as an input and encodes it to the BCD output which is available on the output lines.
```
Comment (YashKumarVerma) [inline docs/MSI/encoder.md:56]: image
https://www.geeksforgeeks.org/encoder-in-digital-logic/

Comment (Dishebh) [reply docs/MSI/encoder.md:56]: Is it the plagiarism issue? If that's so, wouldn't it be possible to use geeksforgeeks as a reference site (like tutorialspoint) and mention it in the references section?

Comment (Dishebh) [reply docs/MSI/encoder.md]: Made some changes accordingly. Pls review, thanks!

File: docs/MSI/encoder.md [hunk 2]
```diff
@@ -50,4 +50,21 @@ That means if D3 = 1 then Y1 Y1 = 11 irrespective of the other inputs. Similarly
 
 <iframe width="100%" height="400px" src="https://circuitverse.org/simulator/embed/762" id="projectPreview" scrolling="no" webkitAllowFullScreen mozAllowFullScreen allowFullScreen> </iframe>
 
+## Decimal to BCD Encoder
+{: .no_toc }
+
+The Decimal to Binary Encoder usually consists of 10 input lines and 4 output lines. Each input line corresponds to the each decimal digit and the output lines generate its corresponding binary code.
```
Comment (YashKumarVerma) [inline docs/MSI/encoder.md:56]: this

File: docs/MSI/encoder.md [hunk 3]
```diff
@@ -50,4 +50,21 @@ That means if D3 = 1 then Y1 Y1 = 11 irrespective of the other inputs. Similarly
...
+
+## Octal to Binary Encoder
+{: .no_toc }
+
+The Octal to Binary Encoder encoder usually consists of 8 inputs lines and 3 outputs lines. Each input line corresponds to the each octal digit and the three output lines generate its corresponding binary code.
```
Comment (YashKumarVerma) [inline docs/MSI/encoder.md:61]: Each input.... binary code

File: docs/MSI/encoder.md [hunk 4]
```diff
@@ -50,4 +50,21 @@ That means if D3 = 1 then Y1 Y1 = 11 irrespective of the other inputs. Similarly
...
+
+## Hexadecimal to Binary Encoder
+{: .no_toc }
+
+The Hexadecimal to Binary Encoder encoder usually consists of 16 inputs lines and 3 outputs lines. Each input line corresponds to the each hexadecimal digit and the three output lines generate its corresponding binary code.
+
```
Comment (YashKumarVerma) [inline docs/MSI/encoder.md:67]: Each input ... binary code

**Comment by Dishebh** [comment]
> it has been picked up directly from https://www.geeksforgeeks.org/encoder-in-digital-logic/
> 
> image
> 
> Good attempt, however, I think I have seen similar content somewhere.
> 
> Please note:
> 
> * You may take the help of links given as references, other digital material, or reference books, but it doesn't mean that you `ctrl+c` and `ctrl+v` everything. Try rephrasing it, explaining the points, or adding valuable information, explanations, illustrations along with.
> * We have our own simulator, where you can realize any circuit as needed. Make use of it to make page interactive.
> * We have various components in the simulator, which can be used for illustrations. Using images from some other sources may invite copyright claims.
> 
> We need to be genuine in our work.
> The CircuitVerse community is here for your help, and types of contributions are valued.

Thank you so much for the insights and the guidelines! Will keep it in mind for later PRs. 
Also, I have tried to fix the plagiarism issue, and added a menu at top (instead of plain text). Pls let me know if the changes are fine.

**Comment by YashKumarVerma** [comment]
Thanks a lot for your contributions.

**Pull request merged**